File: DTC.Api/Dtos/Player/PlayerDto.cs
namespace DTC.Api.Dtos.Player
{
    public class PlayerDto
    {
        public int Id { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string? Nickname { get; set; }

        /// <summary>
        /// Berechneter Name für eine einfache Darstellung im UI (z. B. "Max 'The Power' Mustermann" oder "Max Mustermann").
        /// </summary>
        public string DisplayName => string.IsNullOrWhiteSpace(Nickname)
            ? $"{FirstName} {LastName}"
            : $"{FirstName} '{Nickname}' {LastName}";
    }
}

File: dtc.frontend/src/dtos/player/PlayerDto.ts
import z from "zod";

export const playerDtoSchema = z.object({
  id: z.number().int(),
  firstName: z.string(),
  lastName: z.string(),
  nickname: z.string().nullable(),
  displayName: z.string(),
});

export type PlayerDto = z.infer<typeof playerDtoSchema>;
